skimage/measure/_regionprops.py

```diff
@@ -835,6 +846,133 @@ def regionprops(label_image, intensity_image=None, cache=True):
     return regions
 
 
+def euler_number(image, neighbourhood=8):
+    """Calculate the Euler characteristic in 2D binary image, that characterize
+    the topology of the objects.
+    
+    A neighbourhood configuration is constructed (see variable config), 
+    and a LUT is applied for each configuration.
+
+    Parameters
+    ----------
+    image: (N, M) ndarray, 2D or 3D
+        If image is not binary, all strictly greater than zero values
+        are considered as the object.
+    neighbourhood : 4 or 8 for 2D images, 6 or 26 for 3D images, optional
+        Neighborhood connectivity for object determination. If object is
+        4-connected, then background is 8-connected, and conversely.
+        8-connectivity is used by default for 2D images.
+        26-connectivity is used by default for 3D images.
+
+    Returns
+    -------
+    euler_number : int
+        Euler characteristic
+
+    References
+    ----------
+    .. [1] S. Rivollier. Analyse d’image geometrique et morphometrique par
+           diagrammes de forme et voisinages adaptatifs generaux. PhD thesis,
+           2010. Ecole Nationale Superieure des Mines de Saint-Etienne.
+           https://tel.archives-ouvertes.fr/tel-00560838
+    .. [2] Ohser J., Nagel W., Schladitz K. (2002) The Euler Number of
+           Discretized Sets - On the Choice of Adjacency in Homogeneous
+           Lattices. In: Mecke K., Stoyan D. (eds) Morphology of Condensed
+           Matter. Lecture Notes in Physics, vol 600. Springer, Berlin,
+           Heidelberg.
+
+    Examples
+    --------
+    >>> import numpy as np
+    >>> SAMPLE = np.zeros((100,100,100));
+    >>> SAMPLE[40:60, 40:60, 40:60]=1
+    >>> euler_number(SAMPLE) # doctest: +ELLIPSIS
+    1...
+    >>> SAMPLE[45:55,45:55,45:55] = 0;
+    >>> euler_number(SAMPLE) # doctest: +ELLIPSIS
+    2...
+    >>> SAMPLE = np.array([[0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 0, 1, 1, 0, 0, 0, 0, 0],
+    ...                    [0, 0, 0, 0, 0, 0, 0, 0, 1, 1, 1, 1, 0, 0, 0, 0, 0, 0],
+    ...                    [0, 0, 0, 0, 0, 0, 0, 0, 1, 1, 1, 0, 0, 0, 0, 0, 0, 0],
+    ...                    [0, 0, 0, 0, 0, 0, 0, 0, 1, 1, 1, 0, 0, 0, 0, 0, 0, 0],
+    ...                    [0, 0, 0, 0, 0, 0, 1, 1, 1, 1, 1, 1, 0, 0, 0, 0, 0, 0],
+    ...                    [0, 0, 0, 0, 0, 0, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 0],
+    ...                    [0, 0, 0, 0, 0, 0, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 0],
+    ...                    [1, 0, 1, 0, 0, 1, 1, 0, 1, 1, 0, 0, 1, 1, 1, 1, 1, 0],
+    ...                    [0, 1, 1, 1, 1, 1, 1, 1, 0, 1, 1, 0, 0, 0, 1, 1, 1, 1],
+    ...                    [0, 1, 1, 0, 0, 1, 1, 1, 0, 0, 0, 0, 0, 0, 0, 0, 1, 1]])
+    >>> euler_number(SAMPLE)  # doctest: +ELLIPSIS
```

Comment:
is the ellipsis needed here?

Reply:
Well, as this is my first submission in this kind of big project, I am not used to all tests, ellipsis, etc. Sorry about that. I copied the structure of the measure.perimeter function. I am still not sure if I should keep ELLIPSIS or not.

Reply:
so ELLIPSIS is for the doctests (ie the code in the docstrings is executed by our continuous integration process). An ELLIPSIS means that the test should check that the beginning of the output is correct but forget about what's next. This is very useful when different versions will return a different number of digits for floats for example. However here it's an integer so no need for ELLIPSIS.

Reply:
thanks, I suppressed ELLIPSIS.